Updates related to psc-0.10

zudov opened the pull request.

zudov committed:
Update dependencies

zudov committed:
Define instances using newtype deriving

zudov committed:
Derive Semigroup and Monoid

zudov committed:
Update npm's devDependencies

paf31 commented on src/Data/FormURLEncoded.purs:
```diff
@@ -27,15 +28,11 @@ toArray :: FormURLEncoded -> Array (Tuple String (Maybe String))
 toArray (FormURLEncoded a) = a
```
These could be replaced with a `Newtype` instance now.

zudov replied:
I think we should add a `Newtype` instance, but still have monophonic functions.

I think `Newtype` is very nice with it's higher-order combinators (e.g. `over MyNewType f` vs `MyNewtype <<< f <<< runNewType`), and possibility for abstracting over newtypes (e.g. `StrMap` with polymorphic functions `forall key. Newtype key String`).

However for simple construction/deconstruction I think I'll often be preferring plain functions:
- they make mistakes more wrong looking, in order to construct/deconstruct a type you have to somehow mention it's name, either directly in a function name (`runNewtype`) or through qualified import (`MyNewType.fromArray`).
- they are more interference-friendly `(serialize $ MyNewType.fromArray [...]` vs `serialize (wrap [ ... ] :: MyNewType)` where `serialize :: Serializable a => a -> String`.

zudov committed:
Add `Newtype` instance

zudov committed:
Expose `FormURLEncoded` constructor.

Since we expose the newtype internals through the `Newtype` instance it
makes sense to expose the constructor too.
